### PR TITLE
UPF - Aries Integration

### LINKED ISSUE #30: Translating UPF Protobuf message into chronicles

### PR DESCRIPTION
**Pull request by franklinselva**
closes #30

### CODE REVIEW

File: grpc/server/src/chronicles.rs [hunk 1]
```diff
@@ -329,6 +353,25 @@ fn read_chronicle_template(c: Container, action: ChronicleAs, context: &mut Ctx)
         name.push(arg.into());
```
Comment (franklinselva) [inline grpc/server/src/chronicles.rs:398]: @arbimo `at line 351` The action parameters cannot be converted into `SVar`.

Produces,
```
thread 'main' panicked at 'Variable with empty symbolic domain (note that we do not properly handle optionality in this case)', aries/model/src/model.rs:177:13
```

The file is at [model.rs](https://github.com/franklinselva/aries/blob/protobuf-interface/model/src/model.rs)

File: grpc/server/src/chronicles.rs [hunk 2]
```diff
@@ -385,14 +380,21 @@ fn read_chronicle_template(
...
         let tpe = context
             .model
             .get_symbol_table()
             .types
-            .id_of(&arg)
-            .ok_or_else(|| arg.invalid("Unknown atom"))?;
+            .id_of(&arg_type)
+            .ok_or_else(|| arg.invalid("Unknown argument"))?;
+        println!("base_name: {:?}", base_name);
         let arg = context.model.new_optional_sym_var(tpe, prez, c / VarType::Parameter); // arg.symbol
```
Comment (franklinselva) [inline grpc/server/src/chronicles.rs:398]: @arbimo `at line 398` The action parameters cannot be converted into `SVar`.

Produces,
```
thread 'main' panicked at 'Variable with empty symbolic domain (note that we do not properly handle optionality in this case)', aries/model/src/model.rs:177:13
```

The file is at [model.rs](https://github.com/franklinselva/aries/blob/protobuf-interface/model/src/model.rs)

Comment (arbimo) [reply grpc/server/src/chronicles.rs:398]: Mmmh, there is a know bug (#27) that prevents variables from having an empty domain (even if the variable is not used).
Is there any object declared with this type `tpe` in the domain ?

Comment (arbimo) [reply grpc/server/src/chronicles.rs:398]: For the domain in #30, ti should not : `l_from` has the `Location` type which has 2 insances `l1` and `l2`. So maybe the problem is on the addition of these instances to the symbol table ?

Comment (franklinselva) [reply grpc/server/src/chronicles.rs:398]: For the same problem, the instances and types are available in the symbol table. But the `ContiguousSymbols` for the type `Location` stays the same for both `first = 4` and `after_last = 4` which fails in the `ContiguousSymbols.bounds()` condition (the current type can never pass `first < after_last`).

```rust
pub struct ContiguousSymbols {
    first: usize,
    after_last: usize,
}
```

Comment (franklinselva) [reply grpc/server/src/chronicles.rs:395]: Thank you. I am able to test some supported problems that have not yet been solved. I will try to create more simple examples for testing.

File: grpc/server/src/chronicles.rs [hunk 3]
```diff
@@ -351,7 +403,7 @@ fn read_chronicle_template(
     let start = FAtom::from(start);
 
     let end: FAtom = match action_kind {
-        ChronicleKind::Problem => panic!("unsupported case"),
+        ChronicleKind::Problem => return Err(anyhow!("Problem type not supported")),
```
Comment (arbimo) [inline grpc/server/src/chronicles.rs:409]: The `bail!("Problem...")` macro from anyhow would do the same thing.

Comment (franklinselva) [reply grpc/server/src/chronicles.rs]: Noted.

**Comment by franklinselva** [comment]
Closing this PR due to change in Unified Planning proto definition. The integration is moved to PR(#34)

**Pull request closed without merging**